addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/launch-product.test.js

```diff
@@ -0,0 +1,79 @@
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { iteratee } = require('lodash');
+const WorkflowPayload = require('@aws-ee/workflow-engine/lib/workflow-payload');
+const LaunchProduct = require('../launch-product/launch-product');
+
+describe('LaunchProduct', ()=> {
+      const meta = {};
+      const input = {};
+    
+      const lp = new LaunchProduct({
+        step: { config: {} },
+        workflowPayload: new WorkflowPayload({ meta, input, workflowInstance: { steps: [] } }),
+      });
+
+    describe('checkNamespace', () => {
+        it('Static name should be transformed to start with analysis- and end with number string', async () => {
+            // Build 
+            const resolvedInputParams = [{Key: 'Namespace', Value: 'staticname'}];
+            const datetime = Date.now();
+    
+            // Operate
+            const after = await lp.checkNamespace(resolvedInputParams, datetime);
+    
+            // Check
+            await expect(after[0].Value.startsWith('analysis-')).toBeTruthy() && expect(Number(after[0].Value.split('-').pop())).not.toBe(NaN);
+        });
+    
+        it('Static name that begins with analysis- should be transformed to end with number string', async () => {
+            // Build
+            const resolvedInputParams = [{Key: 'Namespace', Value: 'analysis-staticname'}];
+            const datetime = Date.now();
+    
+            // Operate
+            const after = await lp.checkNamespace(resolvedInputParams, datetime);
+    
+            // Check
+            await expect(Number(after[0].Value.split('-').pop())).not.toBe(NaN);
+        });
+    
+        it('Dynamic name should not be altered', async () => {
+            // Build
+            const datetime = Date.now();
+            const resolvedInputParams = [{Key: 'Namespace', Value: `analysis-${datetime}`}];
+    
+            // Operate
+            const after = await lp.checkNamespace(resolvedInputParams, datetime);
+    
+            // Check
+            await expect(after[0].Value === resolvedInputParams[0].Value).toBeTruthy();
+        });
+    
+        it('Static name that ends with a number sequence should be transformed to start with analysis- and end with number string that is unique', async () => {
+            // Build
+            const resolvedInputParams = [{Key: 'Namespace', Value: 'staticname-2626262626'}];
+            const datetime = Date.now();
+    
+            // Operate
+            const after = await lp.checkNamespace(resolvedInputParams, datetime);
+    
+            // Check
+            const numberSequence = resolvedInputParams[0].Value.split('-').pop();
+            await expect(after[0].Value.startsWith('analysis-')).toBeTruthy() && expect(Number(after[0].Value.split('-').pop())).not.toBe(NaN) && expect(after[0].Value.split('-').pop() === numberSequence).toBeFalsy();
```

Comment:
Would comparing `resolvedInputParams[namespaceIndex]` and `analysis-staticname-2626262626-${datetime}` to be equal be simpler?

Reply:
Had to change the function to use the parameter datetime to do this, but I think that does make sense because it does make this test more straightforward and as you said, does get rid of any unknown function values.